common/persistence/cassandra/nexus_service_store.go

```diff
@@ -61,30 +104,282 @@ func (s *NexusServiceStore) Close() {
 	}
 }
 
-func (s *NexusServiceStore) GetNexusIncomingService(
+func (s *NexusServiceStore) CreateOrUpdateNexusIncomingService(
 	ctx context.Context,
-	name string,
-) (*p.InternalGetNexusIncomingServiceResponse, error) {
-	return nil, serviceerror.NewUnimplemented("Cassandra GetNexusIncomingService is not implemented")
+	request *p.InternalCreateOrUpdateNexusIncomingServiceRequest,
+) error {
+	batch := s.session.NewBatch(gocql.LoggedBatch).WithContext(ctx)
+
+	if request.Service.Version == 0 {
+		batch.Query(templateCreateIncomingServiceQuery,
+			constServicesPartition,
+			rowTypeIncomingNexusService,
+			request.Service.ServiceID,
+			request.Service.Data.Data,
+			request.Service.Data.EncodingType.String(),
+			1,
+		)
+	} else {
+		batch.Query(templateUpdateIncomingServiceQuery,
+			request.Service.Data.Data,
+			request.Service.Data.EncodingType.String(),
+			request.Service.Version+1,
+			constServicesPartition,
+			rowTypeIncomingNexusService,
+			request.Service.ServiceID,
+			request.Service.Version,
+		)
+	}
+
+	if request.LastKnownTableVersion == 0 {
+		batch.Query(templateCreateTableVersion,
+			constServicesPartition,
+			rowTypePartitionStatus,
+			constTableVersionServiceID,
+			1)
+	} else {
+		batch.Query(TemplateUpdateTableVersion,
+			request.LastKnownTableVersion+1,
+			constServicesPartition,
+			rowTypePartitionStatus,
+			constTableVersionServiceID,
+			request.LastKnownTableVersion)
+	}
+
+	previousPartitionStatus := make(map[string]interface{})
+	applied, iter, err := s.session.MapExecuteBatchCAS(batch, previousPartitionStatus)
+
+	if err != nil {
+		return gocql.ConvertError("CreateOrUpdateNexusIncomingService", err)
+	}
+
+	previousService := make(map[string]interface{})
+	iter.MapScan(previousService)
+
+	err = iter.Close()
+	if err != nil {
+		return gocql.ConvertError("CreateOrUpdateNexusIncomingService", err)
+	}
+
+	if !applied {
+		currentTableVersion, err := getTypedFieldFromRow[int64]("version", previousPartitionStatus)
+		if err != nil {
+			return fmt.Errorf("error retrieving current table version: %w", err)
+		}
+		if currentTableVersion != request.LastKnownTableVersion {
+			return fmt.Errorf("%w. Provided table version: %v Current table version: %v",
+				ErrTableVersionConflict,
+				request.LastKnownTableVersion,
+				currentTableVersion)
+		}
+
+		currentServiceVersion, err := getTypedFieldFromRow[int64]("version", previousService)
+		if err != nil {
+			return fmt.Errorf("error retrieving current service version: %w", err)
+		}
+		if currentServiceVersion != request.Service.Version {
+			return fmt.Errorf("%w. Provided service version: %v Current service version: %v",
```

Comment:
Use lower case for Go errors.

```suggestion
			return fmt.Errorf("%w. provided service version: %v, current service version: %v",
```